components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -3111,4 +3116,41 @@ private Property extractPropertyFromMetadata(MetadataConverter metadataConverter
         }
         return propertyWithName;
     }
+
+    /**
+     * Set the confidential status of federated authenticator properties using metadata.
+     * @param identityProvider Identity Provider.
+     */
+    private void setConfidentialStatusFromMeta(IdentityProvider identityProvider)
+            throws IdentityProviderManagementException {
+
+        populateFedAuthConfidentialPropsMap();
+        Arrays.asList(identityProvider.getFederatedAuthenticatorConfigs()).forEach(fedAuthConfig -> {
+            List<String> secretProperties = metaFedAuthConfigMap.get(fedAuthConfig.getName());
+                Arrays.asList(fedAuthConfig.getProperties()).forEach(prop -> {
+                    if (secretProperties != null && secretProperties.contains(prop.getName())) {
+                        prop.setConfidential(true);
+                    }
+                });
+        });
+    }
+
+    /**
+     * Create map of federated authenticator name to list of confidential properties.
+     */
+    private void populateFedAuthConfidentialPropsMap() throws IdentityProviderManagementException {
+
+        if (metaFedAuthConfigMap.isEmpty()) {
+            FederatedAuthenticatorConfig[] metaFedAuthConfigs = getAllFederatedAuthenticators();
+            for (FederatedAuthenticatorConfig metaFedAuthConfig : metaFedAuthConfigs) {
+                List<String> secretProperties = new ArrayList<>();
+                for (Property property : metaFedAuthConfig.getProperties()) {
+                    if (property.isConfidential()) {
+                        secretProperties.add(property.getName());
+                    }
+                }
+                metaFedAuthConfigMap.put(metaFedAuthConfig.getName(), secretProperties);
```

Comment:
Can we create a map and return from this method instead of modifying a static method?